### ISSUE TITLE
Not all env vars set

### ISSUE THREAD

**Issue by xbrianh** [ronf/asyncssh#622]
Slightly extending the [example](https://asyncssh.readthedocs.io/en/latest/index.html#setting-environment-variables) from the docs:
```
async def run_client() -> None:
    async with asyncssh.connect('localhost', password="gh^ds&8e") as conn:
        env = {
            'LANG': "AAA",
            'LC_COLLATE': "BBB",
            'FOO': "BAR",
        }
        result = await conn.run('env', env=env)
        print("AAA" in result.stdout)
        print("BBB" in result.stdout)
        print("BAR" in result.stdout)

try:
    asyncio.get_event_loop().run_until_complete(run_client())
except (OSError, asyncssh.Error) as exc:
    sys.exit('SSH connection failed: ' + str(exc))
```

This outputs:
```
True
True
False
```

Any reason the extra env var `FOO` is not set?

**Comment by ronf** [comment]
SSH servers typically limit the set of environment variables they'll pass through from the client request. So, even though AsyncSSH will pass everything you give it to the server, the server may not pass all of those variables on to the command you are running. The only exception is TERM (terminal type). Everything else is typically disabled by default. There's a note about this in the AsyncSSH documentations. Scroll down on https://asyncssh.readthedocs.io/en/latest/api.html#asyncssh.SSHClientConnectionOptions until you get into the entry for "env" to see a note about this.

On OpenSSH as a server, it much be configured with "AcceptEnv" to accept any environment variables other than TERM from SSH clients.

**Comment by ronf** [comment]
Closing due to inactivity. Feel free to reopen this or open a new issue if you need anything else.

**Comment by Guydada** [comment]
I can confirm this still happens, the env variable is not working properly. I ran the exact same example as shown by @xbrianh and got:
```
False
True
False
```
I checked and my `/etc/ssh/sshd_config` does contain `AcceptEnv LANG LC_*`.  Any ideas? Python version 3.12, asyncssh version 2.17.0

**Comment by ronf** [comment]
I'm not able to reproduce this here with AcceptEnv set to "LANG LC_*" on the server. However, trying the exact same client against the default macOS SSH server does produce False/True/False. My /etc/ssh/sshd_config has no AcceptEnv in it, but there is an AcceptEnv in /etc/ssh/sshd_config.d/100-macos.conf, of "LANG LC_*". So, I'm not sure why it is not setting LANG. In the debug output from AsyncSSH, it is setting it when it makes the request:

```
DEBUG:asyncssh:[conn=0, chan=0]   Env: LANG=AAA
DEBUG:asyncssh:[conn=0, chan=0]   Env: LC_COLLATE=BBB
DEBUG:asyncssh:[conn=0, chan=0]   Env: FOO=BAR
```

When I run my own SSH server on the same system and set "AcceptEnv LANG LC_*" it works correctly with no changes to the AsyncSSH client. So, I don't think the issue is on the AsyncSSH side. Do you have this problem with any environment variables other than LANG, if you adjust AcceptEnv? What OS are you running?

**Comment by ronf** [comment]
I went back tonight at looked at this again, and I can't reproduce the False/True/False error I thought I was seeing with the native macOS server. I think I misread the previous test result.

I'm consistently seeing True/True/False on both the native macOS SSH server and my own installation of OpenSSH, which is the expected result with AcceptEnv set on the server to "LANG LC_*". Also, I'm able to get exactly these same results when using OpenSSH as a client.

**Comment by Guydada** [comment]
Thanks for replying.
I am seeing this behavior when using my client on an Ubuntu 22.04 on WSL, connecting to a server running Linux Ubuntu 22.04, oddly enough as I attached above, I was able to set `LC_COLLATE`, but any other env variable, custom or preset, I could not change and had to workaround this.

**Comment by ronf** [comment]
Just tried to test against Ubuntu 22.04 here, and I saw something interesting. As you mentioned, LANG didn't get picked up correctly from what I set on the client. However, instead of LANG being unset, it was set to "en_US.UTF-8", which is what it gets set to when I open a local terminal window, as if there's something else on the system that's automatically setting this value when starting up new processes (at least in the case of shells).

I'm thinking this might be related to /etc/default/locale, which on my system is currently:

LANG=en_US.UTF-8

When I commented that out, the value I passed in over SSH was set properly, but it looks variables set in this file will override anything passed in via SSH.
